app/controllers/applications_controller.rb

```diff
@@ -79,6 +79,29 @@ def completed_applications
     else 
       redirect_to root_path
     end
+
+  end
+
+  def download_spreadsheet
+
+    applications = Application.all.where(application_status: 'finished')
```

Comment:
Let's just use `Application.where(application_status: 'finished')`. Using `Application.all` will load all of the applications into memory and we don't need them all.

If you want to get ambitious, you can look at model `scopes` and see how they can be applied here. The goal would be to be able to write `Application.finished` - nice and clean.